### ISSUE TITLE
Version 2.0.0

### ISSUE THREAD

**Issue by wmcginty** [BottleRocketStudios/iOS-Hyperspace#39]
There's a large set of changes incoming, this is a placeholder for the incoming 2.0.0 release.

**Comment by tylermilner** [comment]
One other breaking change we might want to get into version 2.0 is the change of `HTTP.Status` to a `RawRepresentable` struct (similar to `HTTP.HeaderKey` and `HTTP.HeaderValue`). I'll work on getting a PR up for that soon.

We can also push out a 2.0 pre-release soon so that our projects that are currently using the new features don't have to point to the release branch.

**Comment by tylermilner** [comment]
I think the only thing that we've got left for this release is the great renaming. I think the general consensus is leaning towards:
`NetworkRequest` -> `Request`
`AnyNetworkRequest<T>` -> `AnyRequest<T>`

**Comment by wmcginty** [comment]
The request rename is merged. Last thing we need to determine is if we want a 2.0-beta or go straight to 2.0. Thoughts @tylermilner @earlgaspard ?

**Comment by earlgaspard** [comment]
beta sounds like it is not done yet. is it done yet?

**Comment by wmcginty** [comment]
For me it means it's done, but it hasn't been thoroughly tested in a "production" environment yet. Unit Tests and examples can only go so far. I'm open to suggestions - RC maybe?

**Comment by earlgaspard** [comment]
I'm for 2.0.0. if we find problems, we can do hot fixes and increase that number. example: 2.0.1
that's my suggestions

**Comment by tylermilner** [comment]
Yeah, I think we're good to go ahead and release as `2.0.0`. I think @earlgaspard's project has been pointing to the release branch for a while now.

**Comment by wmcginty** [comment]
Sounds good. Same as mine actually.

**Comment by tylermilner** [comment]
Just to keep in sync, I think the main thing blocking us from `2.0.0` release is the migration guide.

I was also thinking it might be nice to update the `README` to mention some of the advanced features that were added recently - like the retry/error handling logic. This would mainly point to the docs on those topics, but it would be good to have them mentioned in the main `README` so that the library doesn't seem too primitive upon first glance.

**Comment by wmcginty** [comment]
OK, I can take a look at updating the Readme for the new features if someone wants to take a stab at a basic migration guide.
